feeds/nuget/nuget.go

```diff
@@ -0,0 +1,174 @@
+package nuget
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/ossf/package-feeds/feeds"
+)
+
+const (
+	FeedName           = "nuget"
+	feedURL            = "https://api.nuget.org/v3/index.json"
+	catalogServiceType = "Catalog/3.0.0"
+)
+
+type nugetServiceIndex struct {
+	Services []*nugetService `json:"resources"`
+}
+
+type nugetService struct {
+	URI  string `json:"@id"`
+	Type string `json:"@type"`
+}
+
+type catalog struct {
+	Pages []*catalogPage `json:"items"`
+}
+
+type catalogPage struct {
+	URI      string             `json:"@id"`
+	Created  time.Time          `json:"commitTimeStamp"`
+	Packages []*catalogPageLeaf `json:"items"`
+}
+
+type catalogPageLeaf struct {
+	URI            string    `json:"@id"`
+	CatalogCreated time.Time `json:"commitTimeStamp"`
+	Type           string    `json:"@type"`
+}
+
+type nugetPackageDetails struct {
+	PackageID string    `json:"id"`
+	Version   string    `json:"version"`
+	Created   time.Time `json:"published"`
```

Comment:
One small uncertainty I had was around whether it would be more appropriate to use the `published` or `created` property of the package details here. Per [the documentation](https://docs.microsoft.com/en-us/nuget/api/catalog-resource#package-details-catalog-items):

> The created timestamp is when the package was first received by the package source, which is typically a short time before the catalog item's commit timestamp.

> The published timestamp is the time when the package was last listed.

My understanding is that if a package is delisted (made private), and then relisted - `published` will be updated, but `created` will not be. By using the `published` property here delisting and relisting a package will trigger another event in the package feed. Is this desirable, or should I be using `created` here instead?

Reply:
Hmmm.... I don't know that I have a super strong preference. I see pros/cons.

I think keeping it as the published date is fine for now, since I see this as sort of an append-log of packages that we're pushing out. So in that case, it'd be expected to have multiple records that properly indicate that a package was published more than once.